graphspace/middleware.py

```diff
@@ -3,11 +3,16 @@
 from django.http import HttpResponseRedirect, HttpResponse, QueryDict
 import json
 from graphspace.exceptions import *
+from graphspace.database import *
+from channels.handler import AsgiRequest
 
 
 class SQLAlchemySessionMiddleware(object):
 	def process_request(self, request):
-		request.db_session = settings.db.session()
+		if type(request) == AsgiRequest:
+			request.db_session = Database().session()
```

Comment:
This code wouldn't be immediately clear to me if I didn't have some idea of what was going on already, for the following reasons:
1) Calling the Database() method, even though I don't know where it is coming from. Could you make it more explicit?
2) It's not clear why the distinction is being made between AsgiRequest and the others. Could you add a comment explaining?